Emby.Server.Implementations/WebSockets/WebSocketManager.cs

```diff
@@ -0,0 +1,102 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.WebSockets;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using MediaBrowser.Controller.Net;
+using MediaBrowser.Model.Net;
+using MediaBrowser.Model.Serialization;
+using Microsoft.Extensions.Logging;
+using UtfUnknown;
+
+namespace Emby.Server.Implementations.WebSockets
+{
+    public class WebSocketManager
+    {
+        private readonly IWebSocketHandler[] _webSocketHandlers;
+        private readonly IJsonSerializer _jsonSerializer;
+        private readonly ILogger<WebSocketManager> _logger;
+        private const int BufferSize = 4096;
+
+        public WebSocketManager(IWebSocketHandler[] webSocketHandlers, IJsonSerializer jsonSerializer, ILogger<WebSocketManager> logger)
+        {
+            _webSocketHandlers = webSocketHandlers;
+            _jsonSerializer = jsonSerializer;
+            _logger = logger;
+        }
+
+        public async Task OnWebSocketConnected(WebSocket webSocket)
+        {
+            var taskCompletionSource = new TaskCompletionSource<bool>();
+            var cancellationToken = new CancellationTokenSource().Token;
+            WebSocketReceiveResult result;
+            var message = new List<byte>();
+
+            // Keep listening for incoming messages, otherwise the socket closes automatically
+            do
+            {
+                var buffer = WebSocket.CreateServerBuffer(BufferSize);
```

Comment:
this looks awfully like other websocket handling block up there, why duplicate?

Reply:
Which block? Old commit?

Reply:
Oh, I hadn't looked at the filename. This file is a WIP replacement, but I'm thinking of scrapping it in this PR since it may be too many changes at once.

https://github.com/jellyfin/jellyfin/pull/1010/files#diff-5ff7dd3f2f09dc8b526fc45b5264635fR645

Reply:
What does this mean in the end, can you reduce the copy-pasteness here, or is it the necessary evil?

Reply:
It's a direct replacement hence the copy-pasteness...